### ISSUE TITLE
No SHA

### ISSUE THREAD

**Issue by RGlabs84** [JayDDee/cpuminer-opt#204]
in release notes, v3.6.1 or .2 says SHA support added to m7m, its not working
![nosha](https://user-images.githubusercontent.com/51202846/62073924-aee4e200-b20f-11e9-8c34-97d4205759af.png)

**Comment by JayDDee** [comment]
It has to be compiled for SHA.

**Comment by JayDDee** [comment]
You apparently have a CPU with SHA but not AVX2.I can't find any specs that says the N4200
has SHA but cpuminer says it does based on CPUID.

There is no prebuilt binary for this configuration, The Ryzen build includes SHA but also AVX2.

The N4200 is not a popular CPU and is not good for mining so there is no reason to provide
binaries for this specific architecture.

You might be able to use the ryzen build to mine m7m because there is no AVX2 in that
algo's code. As long as the compiler didn't decide to vectorize any common code you should
be ok. It will be obvious, it will crash if it attempts to execute any AVX2 instructions.

**Comment by JayDDee** [comment]
A bit of unsolicited advice..

If that laptop no longer has any other purpose and you want to take advantage of the SHA
feature for mining (it's basically useless for mining anything else) then dump Windows and
install Ubuntu so you can easilly compile for your CPU's native architecture.

Keep an eye on temperature, laptops aren't designed to run hard.

**Comment by RGlabs84** [comment]
Unfortunately, I use the laptop for work, mostly just web browsing PDF and
printing, damn, the Zen one crashes, I tried that one after reading further
release notes,. Was just hopeful that having the algo and everything
support it would help it's dismissal hashrate.

At RGlabs we try to push everything we can to it's max, having the support
I'm sure would help. It's not a bad proc. Stays turbo around 2.2ghz with
laptop cooler.

Any chance for an older Intel Sha fork?

Your talking to a guy running a Chromebook 3 with win10 using as a low
power web server. Lmao.

My Linux experience is extremely limited, and what limits I have are to
ease of use Fedora

**Comment by JayDDee** [comment]
You may still be using SHA regardless of how cpuminer was compiled or what it says at
startup. 

Unlike other optimizations I use the OpenSSL implementation of sha256. As an external
library it may ignore the application's compile flags and use SHA based only on HW support.

Unfortunately there's no easy way to confirm it.

**Comment by RGlabs84** [comment]
Hmmm, I dont know, honestly, I would prefer that CPUs get as much love as they can afford too, I don’t think its using it based on other observations with other miners, honestly, I don’t know how hard this is, the programming, compiling, etc.  Back when I was a kid a took a few C++ and VB classes, the only thing I remember are #include header.h and cin cout, bahaha. Is it easy to maybe make an extra compile for winblows? For the tune of old and weird intel chips like the n4200, I think some newer cheap laptops feature the n5000 Pentiums, almost like the new gold and silver bullcrap they sling.

I’m honestly always been an AMD guy, ha anyway 3DNow! Can help support mining? All my stuff is either weird new stuff or old decent stuff. I was already formulating a plan for building an older dual Xeon setup and jamming some GPUs in it to have a well rounded personally mining box

Let me know what you think, give it a shot, you’d be surprised how many of us have older crap that we use in this manner, full throttle, pedal down.

Spawn an oldie but goodie kinda thing,

**Comment by JayDDee** [comment]
I'm convinced you already have SHA support. As I said, it uses the OpenSSL library which
was compiled seperately for my native architecture, Ryzen. In testing I noticed I got the same
performance on m7m regardless whether SHA was compiled in or not.

I wouldn't start building for your odd CPU anyway, I always try to keep the number of binaries
to a minimum and there's one more already to be added in the future for AVX512.
